### DISCUSSION TITLE
Some way to set a default callback

### DISCUSSION THREAD

**Discussion by sgued** [slint-ui/slint#4937] [Ideas] [answered]
It appears that it is  currently not possible to set a default callback in a global export or for a  component.

This would be very useful to have some indication (using debug) that a callback has been called when using the viewer. This is especially important for callbacks  that return a value, where the viewer's `on` argument cannot return anything. This would help ensure that the callback is properly called.

For example an app that parses hexadecimal and prints ascii. You would create a `pure callback parse(string) -> string` so that it can be implemented in the host language, but for fast iteration with the viewer it would be appreciable to define a default callback, even if it mocks the actual logic of the callback and just returns some debug value that shows it has been called.

**Comment by ogoffart** [comment, marked as answer]
I think you can do that, for example:
```slint
export component Foo {
 pure callback parse(string) -> string
 parse(x) => { return x; }
}
```
